pkg/webhook/core.oam.dev/v1alpha2/applicationconfiguration/validating_handler_test.go

```diff
@@ -332,37 +347,28 @@ func TestValidateTraitAppliableToWorkloadFn(t *testing.T) {
 				validatingComps: []ValidatingComponent{
 					{
 						compName: "example-comp",
-						component: v1alpha2.Component{ObjectMeta: v1.ObjectMeta{
-							Labels: map[string]string{oam.WorkloadTypeLabel: "TestWorkload0"},
-						}},
 						workloadDefinition: v1alpha2.WorkloadDefinition{
-							TypeMeta: v1.TypeMeta{
-								APIVersion: "unknown.group/v1",
-								Kind:       "TestWorkload1",
-							},
+							ObjectMeta: v1.ObjectMeta{Name: "TestWorkload"},
 							Spec: v1alpha2.WorkloadDefinitionSpec{
 								Reference: v1alpha2.DefinitionReference{
-									Name: "TestWorkload2",
+									Name: "TestWorkload1",
 								},
 							},
 						},
 						validatingTraits: []ValidatingTrait{
 							{traitDefinition: v1alpha2.TraitDefinition{
-								TypeMeta: v1.TypeMeta{
-									APIVersion: "example.com/v1",
-									Kind:       "TestTrait",
-								},
+								ObjectMeta: v1.ObjectMeta{Name: "TestTrait"},
 								Spec: v1alpha2.TraitDefinitionSpec{
-									AppliesToWorkloads: []string{"example.com", "TestWorkload"},
+									AppliesToWorkloads: []string{"example.com", "TestWorkload2"},
 								},
 							}},
 						},
 					},
 				},
 			},
 			want: []error{fmt.Errorf(errFmtUnappliableTrait,
-				"example.com/v1, Kind=TestTrait", "unknown.group/v1, Kind=TestWorkload1", "example-comp",
-				[]string{"example.com", "TestWorkload"})},
+				"TestTrait", "TestWorkload", "example-comp",
+				[]string{"example.com", "TestWorkload2"})},
```

Comment:
I don't see any invalid cases here, please add some cases that don't match applyTo rule and will be blocked by webhook.

Reply:
Sure, I will complete unit tests.